src/shared/mount-util.c

```diff
@@ -490,52 +490,81 @@ int mount_move_root(const char *path) {
         return RET_NERRNO(chdir("/"));
 }
 
-int mount_pivot_root(const char *path) {
+int mount_pivot_root(const char *path, MountAttrPropagationType type) {
```

Comment:
So, why not simply take a mount flag, e.g. MS_PRIVATE?

Reply:
First, because propagation flags aren't really flags. They are an enum that is passed through the flags argument. They are all mutually exclusive. Second, the terminology of these flags is problematic `MS_SLAVE` being the best example. So whenever possible I avoid them.